core/src/main/java/io/kestra/core/runners/Worker.java

```diff
@@ -148,6 +151,7 @@ public class Worker implements Service, Runnable, AutoCloseable {
     public Worker(
```

Comment:
As you've modified the Worker constructor it's highly possible that some plugin tests don't build anymore (e.g.: https://github.com/kestra-io/plugin-aws/blob/41b7e27d9583823f804e84e1ff9008ef188fdf45/src/test/java/io/kestra/plugin/aws/s3/TriggerTest.java#L52)

Maybe it would preferable to now create a new class for wrapping all worker's configs, e.g.:

`record WorkerConfig(String, id, String group, int numThreads, int numRealtimeTriggerThreads){}`

So that, we don't not break tests each time we have a modification.

Also, what about being able to configure the num threads through the `application.yaml` (and other config) ?

Reply:
Oh, I checked the test in the main repo but not on plugins.
I'll have a look, but the way we create the Worker bean in kind of strange.